### PR TITLE
update wide option output

### LINKED ISSUE #22742: Fix 'kubectl get rc' for rcs with more than one container
Commit 53d4b15 moved columns into wide option to make result more readable. However, the patch did not consider more than one container in the rc, e.g.:
```
NAME              DESIRED          CURRENT                                                               AGE
kube-dns-v10      1                1                                                                     5d
                  kube2sky         gcr.io/google_containers/kube2sky:1.12                                          
                  skydns           gcr.io/google_containers/skydns:2015-10-13-8c72f8c                              
                  healthz          gcr.io/google_containers/exechealthz:1.0
```
The desired output shall be probably something like that:
```
NAME              DESIRED          CURRENT                                                               AGE
kube-dns-v10      1                1                                                                     5d
```
and the remaining information in `-owide`.

### PR DESCRIPTION
**Pull request by adohe-zz**
Fixes #22742

### CODE REVIEW

**Comment by AdoHe** [comment]
@bgrant0607 ptal I just add a quick update. I think there are still some little problems:
1. printPodTemplate behavior changed a little, a extra tab is add.
2. kubectl get output is far from perfect, we still have a lot to do, I will think about more and give some better solutions.

**Comment by AdoHe** [comment]
@bgrant0607 would you mind reviewing this?

**Comment by bgrant0607** [comment]
@AdoHe Sorry! This was buried in my email. Please reference the associated issue in the PR description and cc @kubernetes/kubectl on kubectl PRs.

**Comment by bgrant0607** [comment]
cc @janetkuo

**Comment by bgrant0607** [comment]
@AdoHe Could you please paste example output into the PR?

**Comment by AdoHe** [comment]
Ok

File: pkg/kubectl/cmd/cmd_test.go [hunk 1]
```diff
@@ -416,6 +416,59 @@ func ExamplePrintReplicationControllerWithWide() {
...
+		},
+	}
+	err := f.PrintObject(cmd, ctrl, os.Stdout)
+	if err != nil {
+		fmt.Printf("Unexpected error: %v", err)
+	}
+	// Output:
+	// NAME      DESIRED   CURRENT     AGE       CONTAINER(S)   IMAGE(S)    SELECTOR
+	// foo       1         1           10y       foo            someimage   foo=bar
+	//           foo2      someimage
```
Comment (bgrant0607) [inline pkg/kubectl/cmd/cmd_test.go]: What I want is:
```
	// NAME      DESIRED   CURRENT     AGE       CONTAINER(S)   IMAGE(S)    SELECTOR
 	// foo       1         1           10y       foo,foo2            someimage,someimage2   foo=bar
```

1 line

No spaces in comma-separated lists, so columns can be `cut`.

Comment (bgrant0607) [inline pkg/kubectl/cmd/cmd_test.go]: With columns aligned, of course. :-)

Comment (bgrant0607) [inline pkg/kubectl/cmd/cmd_test.go]: We need to eliminate layoutOtherContainers

Comment (bgrant0607) [inline pkg/kubectl/cmd/cmd_test.go]: Or rather convert it to `layoutContainers`, and use it everywhere we print columns for container names and images.

Comment (AdoHe) [inline pkg/kubectl/cmd/cmd_test.go]: Ok I will fix it now.

**Comment by AdoHe** [comment]
@bgrant0607 I think this should be what we want:
```
[tony@localhost kubernetes]$ cluster/kubectl.sh get rc nginx -o wide -L tier --show-labels=true
NAME      DESIRED   CURRENT   AGE       CONTAINER(S)   IMAGE(S)     SELECTOR    TIER      LABELS
nginx     3         3         2s        nginx,perl     nginx,perl   app=nginx   backend   tier=backend
```

**Comment by bgrant0607** [comment]
Yes, thanks! Please notify me once you've pushed the changes.

**Comment by bgrant0607** [comment]
Actually, I see them now.

File: pkg/kubectl/resource_printer.go [hunk 2]
```diff
@@ -714,12 +697,12 @@ func printReplicationController(controller *api.ReplicationController, w io.Writ
 	); err != nil {
 		return err
 	}
+
+	if err := layoutContainers(containers, w, options); err != nil {
```
Comment (bgrant0607) [inline pkg/kubectl/resource_printer.go]: This should only be in wide output?

Comment (bgrant0607) [inline pkg/kubectl/resource_printer.go]: For all the resource types.

File: pkg/kubectl/resource_printer.go [hunk 3]
```diff
@@ -1021,32 +957,37 @@ func printIngress(ingress *extensions.Ingress, w io.Writer, options PrintOptions
...
-		if rules.HTTP == nil {
-			continue
-		}
-		_, err := fmt.Fprintf(w, "%s\t%v\t", extraLinePrefix, rules.Host)
-		if err != nil {
-			return err
-		}
-		if _, err := fmt.Fprint(w, appendLabelTabs(options.ColumnLabels)); err != nil {
-			return err
+	// Lay out all the rules on separate lines if use wide output.
```
Comment (bgrant0607) [inline pkg/kubectl/resource_printer.go]: kubectl get output should always be single-line. We can always print more in describe.

https://github.com/kubernetes/kubernetes/blob/master/docs/devel/kubectl-conventions.md#output-conventions

cc @bprashanth

File: pkg/kubectl/resource_printer.go (hunk 2 above)
Comment (AdoHe) [inline pkg/kubectl/resource_printer.go]: I add a check inside this function, and according to our original code, we put this in wide output.

File: pkg/kubectl/resource_printer.go (hunk 3 above)
Comment (bgrant0607) [inline pkg/kubectl/resource_printer.go]: I suggest not changing `kubectl get ingress` in this PR. We need the container/image fix in ASAP.

Ingress won't be compliant with the convention, but we can fix it in 1.3, if necessary.

**Comment by bgrant0607** [comment]
@AdoHe PTAL at my comments. Thanks!

File: pkg/kubectl/resource_printer.go (hunk 3 above)
Comment (bprashanth) [inline pkg/kubectl/resource_printer.go]: I begrudgingly concede, but one almost always wants to see the host and urls associated with an IP for Ingress. Even if output is getting piped through awk/grep, you're probably looking for more than just the ip.
 
That said, printing just the IP makes implementing kubectl get a lot easier: https://github.com/kubernetes/kubernetes/issues/22837

**Comment by AdoHe** [comment]
In this PR, I don't change the `kubectl get ingress` behaviour, we can leave it to next-candidate, also `layoutContainers` is used for all resource type if necessary.

**Comment by bgrant0607** [comment]
Great! LGTM. Thanks much for doing this on short notice.

**Comment by bgrant0607** [comment]
Updated PR description

**Comment by AdoHe** [comment]
@bgrant0607 Thanks very much.

**Comment by bgrant0607** [comment]
Manually merging.

**Comment by eparis** [comment]
This PR was sucessfully cherry picked in PR #22855
please verify that the release-1.2 branch contains these changes as you would expect and contact @eparis if there appear to be problems.

**Pull request merged**